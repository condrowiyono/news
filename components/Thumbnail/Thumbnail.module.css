.container {
  flex-shrink: 0;
  text-decoration: none;
}

.content {
  margin-top: 12px;
}

.category {
  font-weight: 400;
  font-size: 12px;
  line-height: 160%;
  color: #797D7F;
  margin-bottom: 8px;
  text-transform: capitalize;
}

.title {
  font-weight: 900;
  font-size: 16px;
  line-height: 130%;
  color: #14171A;
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;  
  overflow: hidden;
  margin-bottom: 8px;
}

.total {
  font-weight: 400;
  font-size: 12px;
  line-height: 160%;
  color: #797D7F;
}