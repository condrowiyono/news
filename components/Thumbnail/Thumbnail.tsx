import ImageServer from "@components/Image";
import style from "./Thumbnail.module.css";
import Link from "next/link";
import Skeleton from "@components/Skeleton";

export const SkeletonThumbnail = () => {
  return (
    <div data-testid="title">
      <Skeleton width={156} height={156} borderRadius={8} />
      <div className={style.content}>
        <Skeleton width={96} height={16} />
        <Skeleton width={128} height={24} />
        <Skeleton width={96} height={16} />
      </div>
    </div>
  );
};

const Thumbnail = ({ category, title, total_play, image }: INews) => {
  return (
    <Link href="#">
      <a className={style.container}>
        <ImageServer
          src={image}
          objectFit="cover"
          width={156}
          height={156}
          alt="profile-picture"
          style={{ borderRadius: 8 }}
        />
        <div className={style.content}>
          <div className={style.category}>{category}</div>
          <div className={style.title} data-testid="title">
            {title}
          </div>
          <div className={style.total}>
            <strong>{total_play}</strong> plays
          </div>
        </div>
      </a>
    </Link>
  );
};

export default Thumbnail;
