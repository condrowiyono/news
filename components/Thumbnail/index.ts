import Thumbnail, { SkeletonThumbnail } from "./Thumbnail";

export { SkeletonThumbnail };
export default Thumbnail;