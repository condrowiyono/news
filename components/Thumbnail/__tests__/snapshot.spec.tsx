import { render } from "@testing-library/react";
import Thumbnail, { SkeletonThumbnail } from "@components/Thumbnail";
import { news } from "./Thumbnail.spec";

describe("Snapshot Thumbnail", () => {
  it("renders Thumbnail unchanged", () => {
    const { container } = render(<Thumbnail {...news} />);
    expect(container).toMatchSnapshot();
  });

  it("renders SkeletonThumbnail unchanged", () => {
    const { container } = render(<SkeletonThumbnail />);
    expect(container).toMatchSnapshot();
  });
});
