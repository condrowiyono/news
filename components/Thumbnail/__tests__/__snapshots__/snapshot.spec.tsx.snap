// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Thumbnail renders SkeletonThumbnail unchanged 1`] = `
<div>
  <div
    data-testid="title"
  >
    <div
      class="skeleton"
      data-testid="skeleton"
      style="height: 156px; width: 156px; border-radius: 8px;"
    />
    <div
      class="content"
    >
      <div
        class="skeleton"
        data-testid="skeleton"
        style="height: 16px; width: 96px; border-radius: 4px;"
      />
      <div
        class="skeleton"
        data-testid="skeleton"
        style="height: 24px; width: 128px; border-radius: 4px;"
      />
      <div
        class="skeleton"
        data-testid="skeleton"
        style="height: 16px; width: 96px; border-radius: 4px;"
      />
    </div>
  </div>
</div>
`;

exports[`Snapshot Thumbnail renders Thumbnail unchanged 1`] = `
<div>
  <a
    class="container"
    href="/#"
  >
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27156%27%20height=%27156%27/%3e"
          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
        />
      </span>
      <img
        alt="profile-picture"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="border-radius: 8px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(data:image/gif;base64,R0lGODlhAQABAPAAAL29vf///yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==);"
      />
      <noscript />
    </span>
    <div
      class="content"
    >
      <div
        class="category"
      >
        Bisnis
      </div>
      <div
        class="title"
        data-testid="title"
      >
        Judul
      </div>
      <div
        class="total"
      >
        <strong>
          121212
        </strong>
         plays
      </div>
    </div>
  </a>
</div>
`;
